lib/client/doc.js

```diff
@@ -960,6 +960,21 @@ Doc.prototype._opAcknowledged = function(message) {
     return this.fetch();
   }
 
+  if (message.fixup) {
+    try {
+      this._otApply(message, false);
+    } catch (error) {
+      return this._hardRollback(error);
+    }
+
+    for (var i = 0; i < this.pendingOps.length; i++) {
+      var transformErr = transformX(this.pendingOps[i], message);
+      if (transformErr) return this._hardRollback(transformErr);
+    }
+
+    return;
```

Comment:
I don't know if there's any weirdness here about applying the fixup without incrementing the version? I guess before its ack, the doc is always "ahead" of its version?

Reply:
Fixed this by returning the fixup in the actual ack message